CMSgov/hcgov-design-system
# [LHDM-168] Unauthenticated header

## Summary
Updated the header backing logic and front end to better match the unified header design.

### Added
* Toggles for the login and switch locale links to allow not rendering

### Changed
* The handling of defaultMenuLinks accepts new params to handle the hiding and showing of the unauthenticated links
* defaultMenuLinks has been adjusted to be more of the hub for all links and handles the logout link now as well
* Updated the unit tests and snapshots to accommodate header changes
* Some of the naming on the example page has been updated to not need to repeat the word header everywhere

## How to test
1. Pull down branch. Run `nvm use`, `yarn install`, `yarn start`
2. On the local doc site, navigate to Patterns > Header
3. Play around with the various in versions. Change to Spanish and English. Confirm styles match mocks and functionality is as expected
